test/e2e/etcd_multi_node_test.go

```diff
@@ -0,0 +1,414 @@
+	checkEtcdReady(ctx, cl, logger, etcd)
+}
+
+// cleanUpTestHelperJob ensures to remove the given job in the kubernetes cluster if job exists.
+func cleanUpTestHelperJob(ctx context.Context, cl client.Client, jobName string) {
+	ExpectWithOffset(1,
+		client.IgnoreNotFound(
+			cl.Delete(ctx,
+				&batchv1.Job{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      jobName,
+						Namespace: namespace,
+					},
+				},
+				client.PropagationPolicy(metav1.DeletePropagationForeground),
+			),
+		),
+	).To(Succeed())
+}
+
+// checkJobReady checks k8s job associated pod is ready, up and running.
+// checkJobReady is a helper function, uses Gomega Eventually.
+func checkJobReady(ctx context.Context, cl client.Client, jobName string) {
+	r, _ := k8s_labels.NewRequirement("job-name", selection.Equals, []string{jobName})
+	opts := &client.ListOptions{
+		LabelSelector: k8s_labels.NewSelector().Add(*r),
+		Namespace:     namespace,
+	}
+
+	EventuallyWithOffset(1, func() error {
+		podList := &corev1.PodList{}
+		if err := cl.List(ctx, podList, opts); err != nil {
+			return fmt.Errorf("error occurred while getting pod object: %v ", err)
+		}
+
+		if len(podList.Items) == 0 {
+			return fmt.Errorf("job %s associated pod is not scheduled", jobName)
+		}
+
+		for _, c := range podList.Items[0].Status.Conditions {
+			if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+				return nil
+			}
+
+		}
+		return fmt.Errorf("waiting for pod %v to be ready", podList.Items[0].Name)
+	}, timeout, pollingInterval).Should(BeNil())
+}
+
+// etcdZeroDownTimeValidatorJob returns k8s job which ensures
+// Etcd cluster zero down time by continuously checking etcd cluster health.
+// This job fails once health check fails and associated pod results in error status.
+func startEtcdZeroDownTimeValidatorJob(ctx context.Context, cl client.Client,
+	etcd *v1alpha1.Etcd, testName string) *batchv1.Job {
+	job := etcdZeroDownTimeValidatorJob(etcd.Name+"-client", testName, etcd.Spec.Etcd.ClientUrlTLS)
+
+	logger.Info(fmt.Sprintf("Creating job %s to ensure etcd zero downtime", job.Name))
+	ExpectWithOffset(1, cl.Create(ctx, job)).ShouldNot(HaveOccurred())
+
+	// Wait until zeroDownTimeValidator job is up and running.
+	checkJobReady(ctx, cl, job.Name)
+	logger.Info(fmt.Sprintf("Job %s is ready", job.Name))
+	return job
+}
+
+//getEtcdLeaderPodName returns the leader pod name by using lease
+func getEtcdLeaderPodName(ctx context.Context, cl client.Client, namespace string) (*types.NamespacedName, error) {
+	leaseList := &v1.LeaseList{}
+	opts := &client.ListOptions{Namespace: namespace}
+	ExpectWithOffset(1, cl.List(ctx, leaseList, opts)).ShouldNot(HaveOccurred())
+
+	for _, lease := range leaseList.Items {
+		if lease.Spec.HolderIdentity == nil {
+			return nil, fmt.Errorf("error occurred while finding leader, etcd lease %q spec.holderIdentity is nil", lease.Name)
+		}
+		if strings.Contains(*lease.Spec.HolderIdentity, "Leader") {
```

Comment:
If you see the backup-restore code the convention of memberID is "memberName:<Leader | Member>
So you accurate test should be: `strings.HasSuffix(holderIdentity, ":Leader")`